### PR TITLE
SAM/CFN JSON schema: write settings to global (not workspace) scope

### PR DESCRIPTION
**Pull request by justinmk3**
Problem
-------
Writing settings to workspace scope is noisy.
We missed this in ed8f69c4f5509a4ce53822cb676a16020ba76d58 #1896.

Solution
--------
- Write settings to global scope (user home directory).
- Do not writing settings if they already exist.
- Check the result of getConfiguration(), do not assume that it is valid.

ref #1933

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/cloudformation/activation.ts [hunk 1]
```diff
@@ -58,12 +58,22 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
...
+ * Lifted near-verbatim from the cfn-lint VSCode extension.
  * https://github.com/aws-cloudformation/cfn-lint-visual-studio-code/blob/629de0bac4f36cfc6534e409a6f6766a2240992f/client/src/extension.ts#L56
  */
 function addCustomTags(): void {
-    const currentTags = vscode.workspace.getConfiguration().get<string[]>('yaml.customTags') ?? []
+    const settingName = 'yaml.customTags'
+    const currentTags = vscode.workspace.getConfiguration().get<string[]>(settingName) ?? []
+    if (!Array.isArray(currentTags)) {
+        getLogger().error(
+            'setting "%s" is not an array. SAM/CFN intrinsic functions will not be recognized.',
```
Comment (justinmk3) [inline src/shared/cloudformation/activation.ts:72]: Reminder that we can't trust the result of `getConfiguration().get()` (the type parameter is erased at runtime).

File: src/shared/cloudformation/activation.ts [hunk 2]
```diff
@@ -93,6 +103,9 @@ function addCustomTags(): void {
         '!Split',
         '!Split sequence',
     ]
-    const updateTags = currentTags.concat(cloudFormationTags.filter(item => currentTags.indexOf(item) < 0))
-    vscode.workspace.getConfiguration().update('yaml.customTags', updateTags, vscode.ConfigurationTarget.Workspace)
+    const missingTags = cloudFormationTags.filter(item => !currentTags.includes(item))
+    if (missingTags.length > 0) {
+        const updateTags = currentTags.concat(missingTags)
+        vscode.workspace.getConfiguration().update('yaml.customTags', updateTags, vscode.ConfigurationTarget.Global)
```
Comment (justinmk3) [inline src/shared/cloudformation/activation.ts:109]: Write to Global scope. Also don't update settings if the setting already contains the symbols exist...

File: src/shared/cloudformation/activation.ts (hunk 1 above)
Comment (bryceitoc9) [reply src/shared/cloudformation/activation.ts:72]: Might be worth trying to handle this as an array? Something like https://github.com/aws/aws-toolkit-vscode/blob/7034750338fe7b94217390bf9276ed655dfe005f/src/shared/cloudformation/cloudformation.ts#L857

**Review by bryceitoc9 — APPROVED** [review]

File: src/shared/cloudformation/activation.ts (hunk 1 above)
Comment (justinmk3) [reply src/shared/cloudformation/activation.ts:72]: > Might be worth trying to handle this as an array? Something like

But then we are potentially overwriting the user's settings with something unwanted.

**Pull request merged**